partiql-logical/src/lib.rs

```diff
@@ -1,14 +1,58 @@
 use partiql_value::{BindingsName, Value};
-use petgraph::prelude::StableGraph;
-use petgraph::Directed;
 use std::collections::HashMap;
 
+#[derive(Debug, Hash)]
+pub struct OpId(usize);
+
+impl OpId {
+    pub fn index(&self) -> usize {
+        self.0
+    }
+}
+
+impl PartialEq<Self> for OpId {
+    fn eq(&self, other: &Self) -> bool {
+        self.index() == other.index()
+    }
+}
+
+impl Eq for OpId {}
+
 #[derive(Debug)]
-pub struct LogicalPlan(pub StableGraph<BindingsExpr, (), Directed>);
+pub struct LogicalPlan<'a, T> {
+    nodes: Vec<T>,
+    edges: Vec<(&'a OpId, &'a OpId)>,
```

Comment:
I don't think it's worth the hassle of references for the ids here. Might as well use `Vec<(OpId, OpId)>` and ensure the invariant that the ids must reference existing nodes in the `add_flow` fn.

Reply:
Makes sense; changed in the new commit.